### PR TITLE
error message and explanation for #173

### PR DESCRIPTION
**Pull request by Wikunia**

### CODE REVIEW

File: src/model.jl [hunk 1]
```diff
@@ -30,7 +30,11 @@ function create_root_model!(optimizer::MOI.AbstractOptimizer, jp::JuniperProblem
     if optimizer.nlp_data.has_objective
         obj_expr = MOI.objective_expr(optimizer.nlp_data.evaluator)
         expr_dereferencing!(obj_expr, jp.model)
-        JuMP.set_NL_objective(jp.model, optimizer.sense, obj_expr)
+        try
+            JuMP.set_NL_objective(jp.model, optimizer.sense, obj_expr)
+        catch 
+            error("Have you registered a function? Then please register the function also for Juniper see: https://lanl-ansi.github.io/Juniper.jl/stable/options/#registered_functions::Union{Nothing,Vector{RegisteredFunction}}-[nothing]-1")
+        end
```
Comment (ccoffrin) [inline src/model.jl]: What about the case where they are registering a function for an `NLconstraint`?

Comment (Wikunia) [reply src/model.jl]: True it should be happen here: https://github.com/lanl-ansi/Juniper.jl/blob/c86a191936c007dc6a966656742b2cc762cb64c9/src/model.jl#L58 as well. But the way to fix it is the same.

**Comment by ccoffrin** [comment]
LGTM.

**Comment by ccoffrin** [comment]
Might be good to add a note to the changelog as well.

**Pull request merged**